reana/reana_dev/utils.py

```diff
@@ -335,3 +344,147 @@ def get_prefixed_component_name(component):
+        PYTHON_VERSION_FILE,
+    ]:
+        file_relative_path = run_command(
+            f"git ls-files | grep -w {file_} || true",
+            component,
+            display=False,
+            return_output=True,
+        )
+        if file_relative_path:
+            version_files[file_] = os.path.join(
+                get_srcdir(component=component), file_relative_path
+            )
+
+    return version_files
+
+
+def get_component_current_version_from_file(component):
+    """Get component's current version."""
+    version_files = get_component_version_files(component)
+    version = ""
+    if version_files.get(HELM_VERSION_FILE):
+        with open(version_files.get(HELM_VERSION_FILE)) as f:
+            chart_yaml = yaml.safe_load(f.read())
+            version = chart_yaml["version"]
+
+    elif version_files.get(PYTHON_VERSION_FILE):
+        spec = importlib.util.spec_from_file_location(
+            component, version_files.get(PYTHON_VERSION_FILE)
+        )
+        module = importlib.util.module_from_spec(spec)
+        spec.loader.exec_module(module)
+        version = module.__version__
+
+    elif version_files.get(JAVASCRIPT_VERSION_FILE):
+        with open(version_files.get(JAVASCRIPT_VERSION_FILE)) as f:
+            package_json = json.loads(f.read())
+            version = package_json["version"]
+
+    return version
+
+
+def bump_semver2_version(current_version, part=None):
+    """Bump a semver2 version string.
+
+    :param current_version: current version to be bumped
+    :type current_version: str
+
+    :return: String representation of the next version
+    :rtype: string
+    """
+    if not semver.VersionInfo.isvalid(current_version):
+        click.echo(
+            f"Current version {current_version} is not a valid semver2 version. Please amend it"
+        )
+
+    parsed_current_version = semver.VersionInfo.parse(current_version)
+    next_version = ""
+    if parsed_current_version.build or part == "build":
+        next_version = parsed_current_version.bump_build()
+    elif parsed_current_version.prerelease or part == "prerelease":
+        next_version = parsed_current_version.next_version("prerelease")
+    elif parsed_current_version.patch or part == "patch":
+        next_version = parsed_current_version.next_version("patch")
+    elif parsed_current_version.minor or part == "minor":
+        next_version = parsed_current_version.next_version("minor")
+    elif parsed_current_version.major or part == "major":
+        next_version = parsed_current_version.next_version("major")
+
+    return str(next_version)
+
+
+def bump_pep440_version(current_version, part=None):
+    """Bump a PEP440 version string.
+
+    :param current_version: current version to be bumped
+    :param part: part of the PEP440 version to bump (one of: [major, minor, micro, pre]).
+    :type current_version: str
+    :type part: str
+
+    :return: String representation of the next version
+    :rtype: string
```

Comment:
Note: the function does not bump `dev.YYYYMMDD` into `dev.<TODAY's YYYMMDD>`, it only increments the value by one.  Perhaps make it a special case to be incremented otherwise here, as described in the issue? (Or we can forget it, since we plan to switch to a/b/rc everywhere...)